File: python/difference-of-squares/difference_of_squares.py
def square_of_sum(count):
    sub = 0
    for i in range(1, count+1):
        sub += i
    total = sub**2
    return total
    pass

def sum_of_squares(count):
    sub = 0 
    for i in range(1, count+1):
        sub += i**2
    return sub
    pass

def difference(count):
    return abs(square_of_sum(count) - sum_of_squares(count))
    

File: ruby/difference-of-squares/difference_of_squares.rb
module BookKeeping
	VERSION = 3
end

class Squares

	attr_reader :base
	def initialize(base)
		@base = base
	end

	def sum_of_squares
		sum = 0
		(1..@base).each do |i|
			sum += i**2
		end
		sum
	end

	def square_of_sum
		sum = 0
		(1..@base).each do |i|
			sum += i
		end
		sum**2
	end

	def difference
		square_of_sum - sum_of_squares
	end

end